pytest_ordering/__init__.py

```diff
@@ -56,23 +56,15 @@ def pytest_collection_modifyitems(session, config, items):
         grouped_items.setdefault(order, []).append(item)
 
     if grouped_items:
-        unordered_items = grouped_items.pop(None, None)
-
         sorted_items = []
-        prev_key = 0
-
-        for key, ordered_items in grouped_items.items():
-
-            if unordered_items and prev_key >= 0 and key < 0:
-
-                sorted_items.extend(unordered_items)
-                unordered_items = None
 
-            prev_key = key
+        unordered_items = [grouped_items.pop(None, [])]
 
-            sorted_items.extend(ordered_items)
+        start_list = sorted((i for i in grouped_items.items() if i[0] >= 0), key=lambda x: x[0])
```

Comment:
Can you use `operator.itemgetter(0)` here for the key? Same in line 64.

Also, try to see if this does the same thing: `sorted([i for i in grouped_items if i >= 0], ...)`. Since you're just using the key, you can iterate over the keys instead of two-tuples of key/value.